### PR TITLE
Bug fix for time out

### PR DESCRIPTION
**Pull request by prabhakhar**

### CODE REVIEW

File: modules/engine/lib/engine/http/request.js [hunk 1]
```diff
@@ -298,6 +297,7 @@ function sendHttpRequest(client, options, args, start, timings, reqStart, key, c
             }
         });
         res.on('end', function () {
+            happy = true;
```
Comment (s3u) [inline modules/engine/lib/engine/http/request.js:301]: We need a similar `happy = true` in the 'data' event handler as well.

Comment (prabhakhar) [inline modules/engine/lib/engine/http/request.js:301]: But, I thought only 'end' can confirm the timeout. Suppose we set happy to true on the first chunk, but second chunk timed out, we need to raise timeout error.

Comment (s3u) [inline modules/engine/lib/engine/http/request.js:301]: I think both are needed, but neither is foolproof as node can do

```
self.emit('timeout');
self.destroy(new Error('timeout'));
```

In addition to setting the flag, we ought to retry the request if it is idempotent (https://github.com/ql-io/ql.io/issues/485)

Comment (prabhakhar) [inline modules/engine/lib/engine/http/request.js:301]: Added as suggested. Also, throwing a warn event instead of console log.

**Pull request merged**